### PR TITLE
[stdlib] Added PowerPC 32-bit support

### PR DESCRIPTION
**Pull request by colemancda**
Added patches to cross compile the Swift stdlib for the PowerPC 32-bit architecture.

### CODE REVIEW

**Comment by cnmade** [comment]
Genius， I like your pull request，fantastic

**Comment by CodaFi** [comment]
Compiler changes LGTM.

Would like stdlib’s eyes on this for the rest.

@swift-ci test

**Comment by CodaFi** [comment]
You may want to update the Swift driver here https://github.com/apple/swift-driver/blob/0f9f247c8cc3d225dbbc263b1ec082bf2c8675fe/Sources/SwiftDriver/Jobs/GenericUnixToolchain%2BLinkerSupport.swift

we should already accept the ppc triple family.

File: stdlib/public/SwiftShims/System.h [hunk 1]
```diff
@@ -173,6 +173,11 @@
 #define SWIFT_ABI_ARM64_OBJC_WEAK_REFERENCE_MARKER_VALUE \
   (1<<SWIFT_ABI_ARM64_OBJC_NUM_RESERVED_LOW_BITS)
 
+/*********************************** powerpc ********************************/
```
Comment (xwu) [inline stdlib/public/SwiftShims/System.h:176]:
```suggestion
/************************************ powerpc *********************************/
```

**Comment by buttaface** [comment]
Heh, nice, are you actually building the Swift compiler for 32-bit PowerPC too? If so, not hitting [this C++ static assert that I hit with Android armv7 a couple months ago](https://github.com/buttaface/termux-packages/runs/4753792885?check_suite_focus=true#step:3:7474)?

**Comment by colemancda** [comment]
@buttaface No, I compile a patched Swift 5.5.3 compiler for Aarch64 (Debian on M1 Pro) and cross compile the stdlib using cmake and buildroot. I am building a qemu mac99 image right now. This is also the strategy I used to cross compile Swift 5.5.3 for Armv7. Only 1 patch was needed provided by @uraimo to fix weak references.

**Comment by colemancda** [comment]
I have had to make changes to LLVM that I will be submitting. Mainly in `TargetInfo.cpp`.

**Comment by colemancda** [comment]
Right now I am getting this error:

```
In file included from ../stdlib/public/runtime/Casting.cpp:18:
In file included from ../include/swift/Runtime/Casting.h:20:
In file included from ../include/swift/Runtime/Metadata.h:20:
../include/swift/ABI/Metadata.h:777:50: error: 'swift_context' parameter can only be used with swiftcall or swiftasynccall calling convention
  SWIFT_CC(swift) void(SWIFT_CONTEXT HeapObject *);
```

**Comment by colemancda** [comment]
LLVM patches
https://github.com/apple/llvm-project/pull/4022

**Pull request merged**